### PR TITLE
add-chain: print plasma error message

### PR DESCRIPTION
**Pull request by bitwiseguy**
Fixes https://github.com/ethereum-optimism/superchain-registry/issues/557

Print an error message if any legacy plasma fields are present in the chain's rollup.json when the add-chain command is run. The error message points the user to the docs section that describes how to adjust the rollup.json to use the new `altda` name and config format.

### CODE REVIEW

File: add-chain/config/plasma.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	DACommitmentType           *string             `json:"da_commitment_type"`
+	DAChallengeWindow          *uint64             `json:"da_challenge_window"`
+	DAResolveWindow            *uint64             `json:"da_resolve_window"`
+	UsePlasma                  *bool               `json:"use_plasma"`
+	PlasmaConfig               *interface{}        `json:"plasma_config"`
+}
+
+func (lp *LegacyPlasma) CheckNonNilFields() error {
+	v := reflect.ValueOf(*lp)
+	typeOfT := v.Type()
```
Comment (geoknee) [inline add-chain/config/plasma.go:22]: This variable name jars a bit, how about `typeOfV`?

File: superchain/superchain.go [hunk 2]
```diff
@@ -156,7 +156,8 @@ type AltDAConfig struct {
 	DAChallengeWindow *uint64 `json:"da_challenge_window" toml:"da_challenge_window"`
 	// DA resolve window value set on the DAC contract. Used in altDA mode
 	// to compute when a challenge expires and trigger a reorg if needed.
-	DAResolveWindow *uint64 `json:"da_resolve_window" toml:"da_resolve_window"`
+	DAResolveWindow  *uint64 `json:"da_resolve_window" toml:"da_resolve_window"`
+	DACommitmentType *string `json:"da_commitment_type" toml:"da_commitment_type"`
```
Comment (geoknee) [inline superchain/superchain.go:160]: Do we need to update downstream packages to pick up this change? Or can it wait?

File: add-chain/config/config.go [hunk 3]
```diff
@@ -34,14 +34,29 @@ func ConstructChainConfig(
...
+		return superchain.ChainConfig{}, fmt.Errorf("error unmarshaling legacy plasma json: %w", err)
+	}
+	if err = plasmaConfig.CheckNonNilFields(); err != nil {
+		fmt.Printf(
+			"❌ Detected legacy plasma configuration fields in rollup json file.\n" +
+				"❌   - Please follow the guide at the following url to migrate to the new altda format:\n" +
+				"❌   - https://docs.optimism.io/builders/chain-operators/features/alt-da-mode#breaking-changes-renaming-plasma-mode-to-alt-da-mode\n")
+		return superchain.ChainConfig{}, err
+	}
+
```
Comment (geoknee) [inline add-chain/config/config.go:51]: Nit/Optional: we could extract this into a helper and call it from the test?

**Review by geoknee — APPROVED** [review]
🙏

File: add-chain/config/config.go (hunk 3 above)
Comment (bitwiseguy) [reply add-chain/config/config.go:51]: What do you mean by "call it from the test"?

Comment (bitwiseguy) [reply add-chain/config/config.go:51]: Ahh ok I see. I'll keep that in mind for a potential refactor in a follow-up

**Pull request merged**